packages/fx-core/src/common/tools.ts

```diff
@@ -639,3 +647,52 @@ export function getAllowedAppIds(): string[] {
     OutlookClientId.Web,
   ];
 }
+
+export async function getSideloadingStatus(token: string): Promise<boolean | undefined> {
+  const instance = axios.create({
+    baseURL: getAppStudioEndpoint(),
+    timeout: 30000,
+  });
+  instance.defaults.headers.common["Authorization"] = `Bearer ${token}`;
+
+  let retry = 0;
+  const retryIntervalSeconds = 2;
+  do {
+    try {
+      const response = await instance.get("/api/usersettings/mtUserAppPolicy");
+      let result: boolean | undefined;
+      if (response.status >= 400) {
+        result = undefined;
+      } else {
+        result = response.data?.value?.isSideloadingAllowed as boolean;
+      }
+
+      if (result !== undefined) {
+        sendTelemetryEvent(Component.core, TelemetryEvent.CheckSideloading, {
+          [TelemetryProperty.IsSideloadingAllowed]: result + "",
+        });
+      } else {
+        sendTelemetryErrorEvent(
+          Component.core,
+          TelemetryEvent.CheckSideloading,
+          new SystemError(
+            "UnknownValue",
+            `AppStudio response code: ${response.status}, body: ${response.data}`,
+            "M365Account"
+          )
+        );
+      }
+
+      return result;
+    } catch (error) {
+      sendTelemetryErrorEvent(
```

Comment:
Don't forget to check the GDPR classify status.

Reply:
Good point. Will check in RC release.